web/src/app/schedules/ScheduleOverrideCreateDialog.js

```diff
@@ -33,79 +33,64 @@ const mutation = gql`
   }
 `
 
-export default class ScheduleOverrideCreateDialog extends React.PureComponent {
-  static propTypes = {
-    scheduleID: p.string.isRequired,
-    variant: p.oneOf(['add', 'remove', 'replace']).isRequired,
-    onClose: p.func,
-    defaultValue: p.shape({
-      addUserID: p.string,
-      removeUserID: p.string,
-      start: p.string,
-      end: p.string,
-    }),
-  }
-
-  static defaultProps = {
-    defaultValue: {},
-  }
+export default function ScheduleOverrideCreateDialog(props) {
+  const [value, setValue] = useState({
+    addUserID: '',
+    removeUserID: '',
+    start: DateTime.local()
+      .startOf('hour')
+      .toISO(),
+    end: DateTime.local()
+      .startOf('hour')
+      .plus({ hours: 8 })
+      .toISO(),
+    ...props.defaultValue,
+  })
 
-  constructor(props) {
-    super(props)
-
-    this.state = {
-      value: {
-        addUserID: '',
-        removeUserID: '',
-        start: DateTime.local()
-          .startOf('hour')
-          .toISO(),
-        end: DateTime.local()
-          .startOf('hour')
-          .plus({ hours: 8 })
-          .toISO(),
-        ...props.defaultValue,
+  const [mutate, { loading, error }] = useMutation(mutation, {
+    variables: {
+      input: {
+        ...value,
+        scheduleID: props.scheduleID,
       },
-    }
-  }
+    },
+    onCompleted: props.onClose,
+  })
 
-  render() {
-    return (
-      <Mutation mutation={mutation} onCompleted={this.props.onClose}>
-        {this.renderDialog}
-      </Mutation>
-    )
-  }
+  return (
+    <FormDialog
+      onClose={props.onClose}
+      title={copyText[props.variant].title}
+      subTitle={copyText[props.variant].desc}
+      errors={nonFieldErrors(error)}
+      onSubmit={mutate}
```

Comment:
Wrap this so we don't pass the event args to `mutate`
```suggestion
      onSubmit={() => mutate()}
```